kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/AnalysisExecutor.kt

```diff
@@ -0,0 +1,65 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ex.ProjectManagerEx
+import org.jetbrains.research.ml.kotlinAnalysis.util.getSubdirectories
+import java.nio.file.Path
+
+/** Setup for common IDEA project. For given [project path][projectPath] creates an instance of [project][Project]. */
+val openOrImportSetup: (Path) -> Project = { projectPath -> ProjectUtil.openOrImport(projectPath, null, true) }
```

Comment:
It is a bad practice to add some global variables. Please, change it into:
```
fun execute(projectsDir: Path, setupProject: (Path) -> Project = { projectPath -> ProjectUtil.openOrImport(projectPath, null, true) }) { ... }
```
In my case we use a default value and it is a good practise

Reply:
Yes, you are right! Thanks!